components/script/dom/htmldialogelement.rs

```diff
@@ -70,6 +70,31 @@ impl HTMLDialogElementMethods for HTMLDialogElement {
         *self.return_value.borrow_mut() = return_value;
     }
 
+    // https://html.spec.whatwg.org/multipage/#dom-dialog-show
+    fn Show(&self) {
+        let element = self.upcast::<Element>();
+
+        // Step 1 TODO: Check is modal flag is false
+        if element.has_attribute(&local_name!("open")) {
+            return;
+        }
+
+        // TODO: Step 2 If this has an open attribute, then throw an "InvalidStateError" DOMException.
+
+        // Step 3
+        element.set_bool_attribute(&local_name!("open"), true);
+
+        // TODO: Step 4 Set this's previously focused element to the focused element.
+
+        // TODO: Step 5 Let hideUntil be the result of running topmost popover ancestor given this, null, and false.
+
+        // TODO: Step 6 If hideUntil is null, then set hideUntil to this's node document.
+
+        // TODO: Step 7 Run hide all popovers until given hideUntil, false, and true.
+
+        // TODO: Step 8 Run the dialog focusing steps given this.
```

Comment:
There is some mechanism at https://github.com/servo/servo/blob/650af7db926e9cc070e136f21a56bd47d566d10a/components/script/dom/document.rs#L1105 which seems to be https://html.spec.whatwg.org/multipage/interaction.html#focus-update-steps

The "dialog focusing steps" appear like layer around that mechanism, so perhaps some sort of calling on the existing focusing mechanism would make sense.

Reply:
I don't think just calling into focusing steps directly is that useful on its own because in practice it's normally going to be the wrong behaviour.

In terms of what Servo needs it':

- The [focus delegate](https://html.spec.whatwg.org/multipage/interaction.html#focus-delegate) concept
- autofocus in general seems missing too and that plays into the dialog algorithm.

Happy to make an issue and link to that from the TODO comment?